<img src="./caps/utn.jpg" align="right" width="150" /> 
#### Teoría de los Circuitos II - 2021

# TP2 - Ejercicio 4

## Cálculos utilizando python y simulaciones circuitales.

#### Grupo N°2


Consigna:

<img src="./caps/cons.jpg" align="center" width="800" /> 

<div class="alert alert-block alert-warning">
<b>Comentario:</b> Para todas las simulaciones se trabajó con valores normalizados y se implementaron filtros tipo KHN.</div>

<div class="alert alert-block alert-info">
<b>1)</b> Cálculo de $\epsilon$ y $n$ para Chebyshev.
</div></div>

In [6]:
import numpy as npy
import math as m

alfa_max = 0.4 # dB
alfa_min = 48 # dB
omega_p = 1 # norm omega
omega_s = 3 # norm omega

eps = npy.sqrt(10**(alfa_max/10)-1)

print( 'eps = {:3.3f}  -  eps**2 = {:3.3f}'.format(eps, eps**2) )

for ii in range(1,7):
    
    a = ii * (m.acosh(omega_s))
    b = (npy.cosh(a))**2
    att = 10 * npy.log10(eps**2 * b)
    
    print('alpha_{:d} = {:3.3f} dB'.format(ii,att)) 

eps = 0.311  -  eps**2 = 0.096
alpha_1 = -0.613 dB
alpha_2 = 14.453 dB
alpha_3 = 29.757 dB
alpha_4 = 45.068 dB
alpha_5 = 60.379 dB
alpha_6 = 75.690 dB


<div class="alert alert-block alert-warning">
<b>Comentario:</b> A partir de la plantilla de filtro pasa altos, se plantea una plantila pasa bajos donde se obtuvo: $\epsilon = 0.311$ y $n = 5$.
</div>

<div class="alert alert-block alert-info">
<b>2)</b> Ubicación de polos y ceros: utilizando procedimiento del Schaumann.
</div></div>

In [3]:
import numpy as npy
import math as m

## Parámetros.
nn = 5
ee = 0.311 

## Calculo a.

aa = (1/nn) * m.asinh(1/ee)

print('a = {:3.3f}'.format(aa))
print('')

## Cálculo de ubicación de polos Gk y wk.
gk = []
wk = []

for k in range(1,nn+1):
    this_gk = (-1) * m.sinh(aa) * npy.sin(((2*k-1)*npy.pi) / (2*nn))
    this_wk = m.cosh(aa) * npy.cos (((2*k-1)*npy.pi) / (2*nn))
    gk.append(this_gk)
    wk.append(this_wk)
    print('G_{:d} = {:3.3f} y w_{:d} = {:3.3f}'.format(k,this_gk,k,this_wk))

print('')

wo_k = []
Q_k = []
    
for ii in range(len(gk)):
    this_wok = npy.sqrt((gk[ii]**2) + (wk[ii]**2))    
    this_qk = this_wok / (2 * gk[ii])
    
    wo_k.append(this_wok)    
    Q_k.append(this_qk)
    
    print('Wo_{:d} = {:3.3f} y Q_{:d} = {:3.3f}'.format(ii+1,this_wok,ii+1,this_qk))

a = 0.377

G_1 = -0.119 y w_1 = 1.019
G_2 = -0.312 y w_2 = 0.630
G_3 = -0.386 y w_3 = 0.000
G_4 = -0.312 y w_4 = -0.630
G_5 = -0.119 y w_5 = -1.019

Wo_1 = 1.026 y Q_1 = -4.304
Wo_2 = 0.703 y Q_2 = -1.126
Wo_3 = 0.386 y Q_3 = -0.500
Wo_4 = 0.703 y Q_4 = -1.126
Wo_5 = 1.026 y Q_5 = -4.304


<div class="alert alert-block alert-info">
<b>3)</b> Ubicación de polos y ceros: calculando raíces del polinomio de Chebyshev de orden 5.
</div></div>

In [2]:
import numpy as npy

den = [-24.576,0,-61.44,0,-53.76,0,-19.2,0,-2.4,0,1]

raices = npy.roots(den)

print(raices)

[ 0.1194795 +1.01966983j  0.1194795 -1.01966983j -0.1194795 +1.01966983j
 -0.1194795 -1.01966983j  0.31280139+0.63019061j  0.31280139-0.63019061j
 -0.31280139+0.63019061j -0.31280139-0.63019061j -0.38664379+0.j
  0.38664379+0.j        ]


<div class="alert alert-block alert-warning">
<b>Comentario:</b> Se toman los polos ubicadas del lado izquierdo del plano Z: $(-0.119 \pm j 1.019)$, $(-0.312 \pm j 0.63)$ y $(-0.386)$
</div>

<div class="alert alert-block alert-info">
<b>4)</b> Simulación LTSpice - Chebyshev utilizando cálculo de polos según Schaumann.
</div></div>

<img src="./caps/chebyo.jpg" align="center" width="800" /> 

<img src="./caps/cheby4.jpg" align="center" width="800" /> 

<div class="alert alert-block alert-warning">
<b>Comentario:</b> Para este circuito se utilizó el método detallado en el libro de Schaumann para diseñar el filtro y se colocó una estructura pasiva de order 1 en la primer etapa.
</div>

<div class="alert alert-block alert-info">
<b>5)</b> Simulación LTSpice - Chebyshev utilizando polinomios.
</div></div>

<img src="./caps/cheby1.jpg" align="center" width="800" /> 

<img src="./caps/cheby2.jpg" align="center" width="800" /> 

<div class="alert alert-block alert-warning">
<b>Comentario:</b> Para este caso se planteó el polinomio de Chebyshev de orden 5 y se hallaron las raíces. Luego, para la primer etapa se utilizó una estructura activa integradora de primer orden.
</div>

<div class="alert alert-block alert-info">
<b>6)</b> Simulación LTSpice - Butterworth $\epsilon = 1$
</div></div>

In [9]:
## calculo de n para maxima planicidad

import numpy as np

alfa_max = 1 # dB
alfa_min = 48 # dB
omega_p = 1 # norm omega
omega_s = 3 # norm omega

eps = 1

print( 'eps = {:3.3f}  -  eps**2 = {:3.3f}'.format(eps, eps**2) )

for ii in range(1,11):
    print('alpha_{:d} = {:3.3f} dB'.format( ii, 10*np.log10(1 + eps**2 * omega_s**(2*ii)) ) ) 

eps = 1.000  -  eps**2 = 1.000
alpha_1 = 10.000 dB
alpha_2 = 19.138 dB
alpha_3 = 28.633 dB
alpha_4 = 38.170 dB
alpha_5 = 47.712 dB
alpha_6 = 57.255 dB
alpha_7 = 66.797 dB
alpha_8 = 76.339 dB
alpha_9 = 85.882 dB
alpha_10 = 95.424 dB


<img src="./caps/butter.jpg" align="center" width="800" /> 

<img src="./caps/butter2.jpg" align="center" width="800" /> 